### ISSUE TITLE
`JWKSet` support for `OIDCTokenHandler`

### ISSUE THREAD

**Issue by adzfaulkner** [symfony/symfony#53491]
### Description

We would like to use the symfony/security-http OIDCTokenHandler but the Identity and Access Management system currently employed exposes a JWK Keyset via it's realm certs url as opposed to a single JWK of which the token hander in question only currently supports.

It would be great if the OIDCTokenHandler could also accept an instance of JWKSet to be passed into it's constuctor which will also incur modifications in how the jms is verified.

### Example
```diff
final class OidcTokenHandler implements AccessTokenHandlerInterface
{
    use OidcTrait;

    public function __construct(
        private Algorithm $signatureAlgorithm,
-       private JWK$jwk,
+       private JWK|JWKSet $jwk,
        private string $audience,
        private array $issuers,
        private string $claim = 'sub',
        private ?LoggerInterface $logger = null,
        private ClockInterface $clock = new Clock()
    ) {
    }
}
```

**Comment by chalasr** [comment]
~~Possible duplicate of https://github.com/symfony/symfony/issues/50434~~

**Comment by chalasr** [comment]
I didn't mean to close as I'm not entirely sure it's the same feature request. Can you please confirm @adzfaulkner?

**Comment by adzfaulkner** [comment]
@chalasr I can confirm it's not a duplicate as the suspected related ticket refers to downloading a single key from a certs URL as opposed to my request which is JWKSet support to be added to the underlying Token Handler

**Comment by OskarStark** [comment]
I made the example in the PR header a diff to be more readable, can you confirm my change is correct? Thanks

**Comment by adzfaulkner** [comment]
@OskarStark spot on thank you :-)

**Comment by louismariegaborit** [comment]
I worked few months ago on this #51665. 😉

**Comment by louismariegaborit** [comment]
I think this issue was resolved by #53682.

**Comment by adzfaulkner** [comment]
@louismariegaborit your assertion looks correct.  Hence issue has been closed. Cheers for the hard work to make it happen!
